indra/assemblers/cag_assembler.py

```diff
@@ -82,16 +82,22 @@ def make_model(self, grounding_threshold = None):
 
             # Edge is solid if both nodes have polarity given
             linestyle = 'solid' if has_both_polarity else 'dotted'
-            targetArrowShape, linecolor = ('tee', 'maroon') if \
-                    (has_both_polarity and
-                     s.subj_delta['polarity'] != s.obj_delta['polarity']) \
-                else ('circle', 'green')
+            if has_both_polarity:
+                same_polarity = s.subj_delta['polarity'] == s.obj_delta['polarity']
+                if same_polarity:
+                    targetArrowShape, linecolor = ('circle', 'green')
+                else:
+                    targetArrowShape, linecolor = ('tee', 'maroon')
+            else:
+                targetArrowShape, linecolor = ('triangle', 'maroon')
+
 
             # Add edge to the graph with metadata from statement
             if s.evidence:
                 provenance = s.evidence[0].annotations.get('provenance', {})
             else:
-                provenance = {}
+                provenance = [{}]
+            provenance[0]['text'] = s.evidence[0].text
```

Comment:
This line seems to assume that there is an evidence list and that the first element of that has a text entry - which is almost certainly the case except if there's a bug somewhere. But then the previous if-else block could also assume that s.evidence has content without checking.